src/manifolds/Stiefel.jl

```diff
@@ -318,3 +318,26 @@ i.e. `(n,k)`, which is the matrix dimensions.
 @generated representation_size(::Stiefel{n,k}) where {n,k} = (n, k)
 
 show(io::IO, ::Stiefel{n,k,F}) where {n,k,F} = print(io, "Stiefel($(n), $(k), $(F))")
+
+"""
+    uniform_distribution(M::Stiefel, p)
+
+Uniform distribution on given [`Stiefel`](@ref) `M`. Generated points will be of
+similar type as `p`.
+"""
+function uniform_distribution(M::Stiefel, p)
+    d = RandnMatrix(size(p)...)
+    return ProjectedPointDistribution(M, d, project!, p)
+end
+
+struct RandnMatrix <: ContinuousMatrixDistribution
+    n::Integer
+    m::Integer
+end
+Base.size(d::RandnMatrix) = (d.n,d.m)
+Base.size(d::RandnMatrix, i) = i::Integer <= 2 ? size(d)[i] : 1
+Distributions._rand!(rng::AbstractRNG, d::RandnMatrix, A::AbstractMatrix) = randn!(rng,A)
+Distributions._logpdf(d::RandnMatrix, x::AbstractArray) = sum(x) do xij
+    dij = Distributions.Normal(zero(xij),one(xij))
+    return Distributions.logpdf(dij,xij)
+end
```

Comment:
Because this is just a multivariate normal variable reshaped into a matrix you should be able to do this:
```suggestion
function Distributions._logpdf(d::RandnMatrix, x::AbstractArray)
    dim = prod(size(d))
    σ = one(eltype(x))
    dvec = MvNormal(dim, σ)
    return Distributions._logpdf(dvec, vec(x))
end
```
This is nice because the `MvNormal(::Integer, ::Real)` signature makes a `ZeroMeanIsoNormal`, which leverages all the efficiencies of having a zero mean and scalar covariance.

Reply:
Good point, incorporated!